executor/src/main/java/com/redhat/service/smartevents/executor/ExecutorsService.java

```diff
@@ -30,10 +26,9 @@ public class ExecutorsService {
     @Incoming(EVENTS_IN_CHANNEL)
     public CompletionStage<Void> processBridgeEvent(final Message<String> message) {
         try {
-            CloudEvent cloudEvent = CloudEventUtils.decode(message.getPayload());
             Executor executor = executorsProvider.getExecutor();
             try {
-                executor.onEvent(cloudEvent);
+                executor.onEvent(message.getPayload());
```

Comment:
We should move in the opposite direction: I would add the CloudEvents parsing here and preserve `executor.onEvent(CloudEvents)` method.
The whole system is designed to manage CloudEvents and in the future RHOC is going to do the conversion for us so at that point we will probably going to have only CloudEvents everywhere.

My suggestion is to check if this a valid CloudEvents, if not, "convert" it as CloudEvents (adding missing metadata) here

Reply:
I refactored everything to use `CloudEvent`s both in source and sink processors as you suggested.
This had two additional major benefits:
* Filtering is now consistent between both type of processors, as they always deal with a `CloudEvent` structure.
* The special `SourceAction` internal action is not needed anymore and standard `Webhook` action can be used to resolve `Source`s. Thus, I deleted it completely.